pkg/shared/clients/nats/nats_client_test.go

```diff
@@ -0,0 +1,141 @@
+package nats
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/nats-io/nats-server/v2/server"
+	"github.com/nats-io/nats.go"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+
+	natstestserver "github.com/nats-io/nats-server/v2/test"
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+)
+
+func TestNewNATSClient(t *testing.T) {
+	// Setting up environment variables for the test
+	os.Setenv(dfv1.EnvISBSvcJetStreamURL, "nats://localhost:4222")
+	os.Setenv(dfv1.EnvISBSvcJetStreamUser, "user")
+	os.Setenv(dfv1.EnvISBSvcJetStreamPassword, "password")
+	defer os.Clearenv()
+
+	log := zap.NewNop().Sugar()
+
+	ctx := logging.WithLogger(context.Background(), log)
+
+	client, err := NewNATSClient(ctx)
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+
+	// Cleanup
+	client.Close()
+}
+
+func TestNewNATSClient_Failure(t *testing.T) {
+	// Simulating environment variable absence
+	os.Clearenv()
+
+	log := zap.NewNop().Sugar()
+	ctx := logging.WithLogger(context.Background(), log)
+
+	client, err := NewNATSClient(ctx)
+	assert.Error(t, err)
+	assert.Nil(t, client)
+}
+
+func RunJetStreamServer(t *testing.T) *server.Server {
```

Comment:
What don't we use the existing one?

Refer to https://github.com/numaproj/numaflow/blob/main/pkg/isb/stores/jetstream/reader_test.go#L135-L136